internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -364,6 +374,90 @@ func (r *NumaflowControllerRolloutReconciler) getResourceOperations() (kubeUtil.
 	return ops, cleanup, nil
 }
 
+func getDeploymentCondition(status appsv1.DeploymentStatus, condType appsv1.DeploymentConditionType) *appsv1.DeploymentCondition {
+	for _, cond := range status.Conditions {
+		if cond.Type == condType {
+			return &cond
+		}
+	}
+	return nil
+}
+
+func (r *NumaflowControllerRolloutReconciler) processNumaflowControllerStatus(ctx context.Context, controllerRollout *apiv1.NumaflowControllerRollout) error {
+	numaLogger := logger.FromContext(ctx)
+
+	// Try to get the Numaflow Controller Deployment
+	deplObj := kubernetes.GenericObject{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "numaflow-controller",
+			Namespace: controllerRollout.Namespace,
```

Comment:
Is there a way to avoid hardcoding the strings? Are they available in the installed CRDs?

Reply:
There is already a [constant](https://github.com/numaproj/numaplane/blob/main/internal/controller/numaflowcontrollerrollout_controller.go#L63) for it, I forgot to use it. Will fix with my next PR. Thanks for catching this.